src/globus_sdk/_sphinxext.py

```diff
@@ -208,7 +208,7 @@ def gen_rst(self):
         base_url = self.options.get("base_url", default_base_url)
         relative_link = self.options["ref"]
         yield (
-            f"See `{message} <{base_url}/{relative_link}>`_ in the "
+            f"See `{message} <{base_url}/{relative_link}>`__ in the "
```

Comment:
This is a pretty obscure change, and I meant to get back to it to add a comment with a reference. I won't resolve this thread until I do so.

Here's how I understand the issue, but I haven't found the docutils docs on this, so I may be missing important details:

`_` vs `__` produces slightly different results, and sphinx and docutils both exhibit the relevant behavior regarding this change -- it is part of the spec, not a sphinx-specific nuance. The way in which it comes up for us, however, derives from sphinx behaviors, in particular autodoc.

The trouble arises with "explicit target names" which are produced in RST from autodoc. The common case I'll point at would be something like this:
```python
def frobulate(self):
    """
    See docs on how to `frobulate <https://example.org/frobulate>`_.
    """
```

Because `frobulate` is both the method name and the target name for the link, there's a conflict.
Autodoc produces a target named `frobulate` for the method and we have one named `frobulate` in the form of our link. A double-underscore `__` trailer states that we do not wish to make the link into a target, which resolves the "duplicate explicit target name" warning (we use `-W`, so it's an error) we see from this.

I'm a little bit fuzzy on details of what a "target" is in rst, but I think it's anything you can point at with a `:ref:`.
I need to try again to find the doc from docutils -- not sphinx! -- which will explain this, so that we can refer to that for a better explanation.

Reply:
I was able to track this down a bit further after writing this up. (It didn't take long; but I had trouble last time I tried. I think I've refined my searches a bit...)

There's a new comment here to try to explain this well, along with links out to docutils. In particular, [this section](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#hyperlink-targets) gave me some more clarity (after I stared at it until I went cross-eyed) on what is meant by a "target". It seems to be the docutils/rst term, generically, for a link target either internal to the documentation tree or external to the internet. So anything we'd link to with an href -- an anchor link within a page, a page on the web, etc -- is a "target".

Reply:
If that's part of docutils, it's unclear to me how this is different from [an anonymous hyperlink](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#anonymous-hyperlinks). Anonymous hyperlinks should look something like:

```rst
I like `eating hamburgers on Tuesdays`__.

..  __: https://example.com
```

Reply:
Shoot, of course I responded before seeing your comment. We really need to establish a comment response ordering algorithm...

Reply:
GitHub enforces ordering for us. It's really conflict resolution we need. I bet there's some good prior art in git for this. :stuck_out_tongue:

Reply:
Haha! Hopefully!